Source/towners.cpp

```diff
@@ -420,22 +385,16 @@ void TalkToBarOwner(PlayerStruct &player, TownerStruct &barOwner)
 	}
 }
 
-void TalkToDeadguy(PlayerStruct &player, TownerStruct &deadguy)
+void TalkToDeadguy(PlayerStruct &player, TownerStruct & /*deadguy*/)
 {
 	if (quests[Q_BUTCHER]._qactive == QUEST_ACTIVE && quests[Q_BUTCHER]._qvar1 == 1) {
-		deadguy._tbtcnt = true;
-		deadguy._tTalkingToPlayer = &player;
 		quests[Q_BUTCHER]._qvar1 = 1;
 		player.PlaySpecificSpeach(8);
-		deadguy._tMsgSaid = true;
 		return;
 	}
 
 	if (quests[Q_BUTCHER]._qactive == QUEST_DONE && quests[Q_BUTCHER]._qvar1 == 1) {
 		quests[Q_BUTCHER]._qvar1 = 1;

```

Comment:
This also seems redundant and can be removed. Checks if `quests[Q_BUTCHER]._qvar1 == 1`, then sets it to 1.

Reply:
So much more redundant an other jank in the quest logic as well :(

Fixed most of it except for the hellfire portion.